src/manifest/extensions/editor/EditorExtension.ts

```diff
@@ -0,0 +1,186 @@
+import { EventType } from '../../../sdk/logging/EventType';
+import { LogLevel } from '../../../sdk/logging/LogLevel';
+import { OutreachContext } from '../../../context/OutreachContext';
+import logger from '../../../sdk/logging/Logger';
+import { UserContextKeys } from '../../../context/keys/UserContextKeys';
+import { ClientContextKeys } from '../../../context/keys/ClientContextKeys';
+import { OrganizationContextKeys } from '../../../context/keys/OrganizationContextKeys';
+import { ShellExtensionType } from '../shell/ShellExtensionType';
+import { EditorRegion } from './EditorRegion';
+import { EditorSettings } from './EditorSettings';
+import { ExtensionHost } from '../ExtensionHost';
+
+export class EditorExtension extends Extension {
+  /**
+   * In this section, the addon author defines a list of predefined context information that addon needs from Outreach
+   * to be sent during the initialization process.
+   * It is a string array of predefined Outreach properties describing attributes of the Outreach user loading the addon.
+   *
+   * @see https://github.com/getoutreach/extensibility-sdk/blob/master/docs/manifest.md#context
+   * @see https://github.com/getoutreach/extensibility-sdk/blob/master/docs/context.md
+   *  @type {(UserContextKeys | ClientContextKeys | OrganizationContextKeys)[]}
+   * @memberof EditorExtension
+   */
+  public context: (UserContextKeys | ClientContextKeys | OrganizationContextKeys)[];
+
+  /**
+   * Definition of addon host
+   *
+   * @see https://github.com/getoutreach/extensibility-sdk/blob/master/docs/manifest.md#host
+   * @type {ExtensionHost}
+   * @memberof EditorExtension
+   */
+  public host: ExtensionHost;
+
+  /**
+   * Type property defines the type of tab extension
+   * @see https://github.com/getoutreach/extensibility-sdk/blob/master/docs/manifest.md#type
+   * @type {ShellExtensionType.EDITOR}
+   * @memberof EditorExtension
+   */
+  public type: ShellExtensionType.EDITOR = ShellExtensionType.EDITOR;
+
+  /**
+   * Optional properties which defines the editor hosting contexts
+   * in which extension should be loaded (eg. only on a sequence and tasks pages)
+   *
+   * If undefined, extension will be loaded in all of the regions.
+   *
+   * @type {EditorRegion[]}
+   * @memberof EditorExtension
+   */
+  public regions?: EditorRegion[];
+
+  /**
+   * Optional settings of the editor extension
+   *
+   * @type {EditorSettings}
+   * @memberof EditorExtension
+   */
+  public settings?: EditorSettings;
+
+  /**
+   * Initialize Outreach context with tab extension contextual information.
+   *
+   * @param {OutreachContext} context
+   * @return {*}  {boolean}
+   * @memberof EditorExtension
+   */
+  init(context: OutreachContext): boolean {
+    let modified = false;
+
+    try {
+      // 1. copy url search parameters to context urlParams
+      const url = new URL(this.host.url!);
+      const searchParams = new URLSearchParams(url.search);
```

Comment:
I believe `url.search` is already an instance of `URLSearchParams` so this step is not needed.